### ISSUE TITLE
Ender 3 Pro support

### ISSUE THREAD

**Issue by tloki** [Creality3DPrinting/Ender-3#23]
As of today (October 26th) Ender 3 Pro does not include same support as its cheaper counterpart. I would very much like to see some firmware updates, profiles etc.

The stock firmware has some "Pro" annotations so I guess that it is different from non-pro one. Even on the printers LCD display it says Ender 3 "Pro".

The profiles that I came by on facebook group do not fit well for me, I suppose it's because of magnetic bed or something, would be nice to have some go-to parameters, as stock ones are horrible to the point that you can't print anything (lots of problems...)

Is this a milestone for Creality, or did I do wrong when I bought Ender 3 Pro version...?

**Comment by bytespider** [comment]
@tloki I doubt your printing issues are relating to the magnetic bed, I print fine with it.
If your slicer is Cura, check the printer set up as there **was** a bug where the filament width was set to 2.85 and not 1.75mm

**Comment by swilkens** [comment]
The "Stock" Ender 3 Pro firmware is listed on the creality website:
https://www.creality3d.cn/download/firmware_c0001

You will notice that this is a fairly old version of Marlin, it will not even compile with an up to date glib.

@sexycyborg would it be possible to get the current Ender 3 Pro firmware sources published here as well? Or are the essentially the same as the Ender 3, which is what I would expect.

**Comment by tloki** [comment]
The difference is (at least) in the screen text showing "Pro" appendix. Although, the more I think about it, the less it makes sense for them to maintain two different firmwares and publish only one, therefore it probably is only a screen text (which is easy to change...)

**Comment by swilkens** [comment]
Considering it's the same hardware for most of their printers, I would expect that they use the same acceleration, feedrate, jerk, steps etc settings for most of their models. It seems likely that there is little difference, yes. 

But it would be nice to have the sources for the factory default firmware nontheless :)

**Comment by sexycyborg** [comment]
I'm going to be reviewing the Pro in the next 1-2 weeks, I'll look for issues and include a request to keep the firmware version on the website more up to date with my recommendations to Creality.

**Comment by swilkens** [comment]
> I'm going to be reviewing the Pro in the next 1-2 weeks, I'll look for issues and include a request to keep the firmware version on the website more up to date with my recommendations to Creality.

Excellent, I hope they are able to provide an updates source here on github or on their website.

While you are reviewing the printer, perhaps you might also look at the default homing speeds on the creality printers. It seems that some users of the "Pro" printer are finding that the default homing speed is so high that it causes the endstops to fully bottom out. See for information:
https://github.com/MarlinFirmware/Marlin/issues/12074

**Comment by NachoFoot** [comment]
> I'm going to be reviewing the Pro in the next 1-2 weeks, I'll look for issues and include a request to keep the firmware version on the website more up to date with my recommendations to Creality.

Can you provide a drawing of the Ender 3 Pro Y Carriage as well, please?

**Comment by sexycyborg** [comment]
> Can you provide a drawing of the Ender 3 Pro Y Carriage as well, please?

I'm not sure, that might comes under Open Sourcing a second hardware design, that's a hard sell. Will work on it.

**Comment by swilkens** [comment]
Hey there @sexycyborg 

I've noticed that Creality has provided a more up to date version of the firmware sources for both the Ender 3 and the Pro model, many thanks! I hope Creality will continue to provide firmware sources for their printers, excellent.

Thanks!

**Comment by tloki** [comment]
Are you talking about these or did they put it somewhere else?

https://github.com/Creality3DPrinting/Ender-3/tree/master/Ender-3%20Firmware%20(Marlin)/Ender-3%20(includes%20power%20failure%20resume-%20English)

**Comment by swilkens** [comment]
For some reason, the ender 3 source is 1.1.6 but the ender 3 pro source appears to be far older.

**Comment by DisasteR** [comment]
Any news about that ?

**Comment by neoOpus** [comment]
Same here waiting for a new firmware since a while and I don't want to risk burning my house because of thermal overrun... Please make the updates and include some of the newest security features of Marlin.

**Comment by Sonnenwerk** [comment]
Hi, dies the Ender pro Haß the Dame Branch aß the Ender at all ?@sexycyborg

**Comment by alexarregoces** [comment]
Does anybody know any news from this? we really need to be able to install the stock firmware for Ender 3 "PRO" which is not the same as ender 3. The leveling auto home height is different and the power lost recovery doesn't work if you use the ender 3 firmware on the ender 3 pro.

**Comment by jpcuzzourt** [comment]
I'm planning to recompile and flash Marlin for my Ender 3 Pro.
I want to be sure I have the correct _Bootscreen.h file for the Pro, but I haven't been able to locate it.
I'm assuming the image data is in the .hex file they provide at their site for the firmware download, but I'm not up to locating it and extracting it tonight, and preferably never. (But I will try if I don't find it elsewhere.)
A regular Ender 3 _Bootscreen.h file is present in the example configurations for Creality at the Marlin github repository.

Is the Ender 3 Pro _Bootscreen.h file available to anyone's knowledge?

**Comment by tloki** [comment]
I may just dump my factory Ender3 firmware before upgrading to a new one, will do just that in upcoming weeks. My version does have resume on poweroff option, but no bed leveling whatsoever. Have no idea which version it is.

**Comment by robthepyro** [comment]
Attached is the firmware from a recent Ender 3 Pro that I dumped earlier. M503 says Marlin 1.1.6 compiled 9/2018 
This should be a good firmware for those looking for a stock working Ender 3 Pro firmware to flash. 


[Ender3Pro1.1.6Dump.zip](https://github.com/Creality3DPrinting/Ender-3/files/3355738/Ender3Pro1.1.6Dump.zip)

**Comment by alexarregoces** [comment]
Awesome! Thank you so much

**Comment by prettypegagirl** [comment]
Thank you robthepyro, quick question is this the one that says "SD card" instead of "TF Card"?

**Comment by robthepyro** [comment]
I believe this is the "TF Card" version 
I'd have to check when I'm with the board again, however. 

What's the difference?

**Comment by prettypegagirl** [comment]
Honestly no idea what the difference is, I bought my pro with the "sd card" version, but just replaced the motherboard and now I got the "TF Card" version. My main issue is the font of the TF card version compared to the SD card version. I found the old one easier to read, I liked the beep when I pressed the button to select something and now that is gone.

I'm quite new to this so not even sure how and if it would be possible to copy the other firmware from the old board?

**Comment by wjsteele** [comment]
TF and SD are the same thing. Chinese companies generally only use the TF wording. A real SD card supports encrypted data, but no one really uses them for that... But there SD consortium used the TF (TransFlash) physical format for their version. The little 8 bit processors don't use the encryption, or even support it, so if you use an SD card in it, it uses the unencrypted TF format.

On Jul 6, 2019 12:41 AM, prettypegagirl <notifications@github.com> wrote:

Honestly no idea what the difference is, I bought my pro with the "sd card" version, but just replaced the motherboard and now I got the "TF Card" version. My main issue is the font of the TF card version compared to the SD card version. I found the old one easier to read, I liked the beep when I pressed the button to select something and now that is gone.

I'm quite new to this so not even sure how and if it would be possible to copy the other firmware from the old board?

**Comment by batesman1** [comment]
So...There's this now.

https://www.creality.com/info/ender-3-ender-3-pro-ender-5-firmware-source-code-upgraded-i00025i1.html

**Comment by cdub3d** [comment]
I am just finding this, after going round and round with someone @ creality on facebook messenger for 6 days so far, trying to get a copy of 1.1.8 firmware to help a newer member of our hobby with one of these printers and the silent board option. The links above seem to provide links to actual 1.1.6.1 source code, and valid firmware hex files, unless you need the 1.1.8 option for Ender 3 Pro with the silent board with 2208's.  (I did find a link to the Chinese version of Ender 5 Pro 1.1.8 with silent board and 2208's, but not ender 3 pro.) 

They even have a sentence on the page for this, "If you Ender 3 pro is with our silent TMC2208 mother board, pls check this link to download the 1.1.8 TMC2208 Ender3 pro firmware: https://www.creality3d.cn/creality3d-ender-3-pro-3d-printer-p00251p1.html "

Two different Ender 3 Pro links, https://www.creality3d.cn/creality3d-ender-3-pro-p00251p1.html or https://www.creality3d.cn/creality3d-ender-3-pro-3d-printer-p00251p1.html
Unfortunately if you click on either link you end up with the same file: "Ender-3 Pro1.1.6.2 resume printing.hex"
Hopefully they will get it posted soon. Meantime, if anyone has the 1.1.8 silent board for an ender 3 pro and knows how to use avrdude or whatever other means to dump a copy of the firmware, I would appreciate it. I enjoy helping folks, but I really don't want to buy a brand new board I don't need just to do that. :)

**Comment by zero4281** [comment]
Marlin is open source.
https://github.com/MarlinFirmware/Marlin

You should definitely consider compiling it yourself.  You can find the correct config file in the zip file linked here: https://github.com/MarlinFirmware/Marlin/tree/2.0.x/config

**Comment by zero4281** [comment]
> I am just finding this, after going round and round with someone @ creality on facebook messenger for 6 days so far, trying to get a copy of 1.1.8 firmware to help a newer member of our hobby with one of these printers and the silent board option. The links above seem to provide links to actual 1.1.6.1 source code, and valid firmware hex files, unless you need the 1.1.8 option for Ender 3 Pro with the silent board with 2208's. (I did find a link to the Chinese version of Ender 5 Pro 1.1.8 with silent board and 2208's, but not ender 3 pro.)
> 
> They even have a sentence on the page for this, "If you Ender 3 pro is with our silent TMC2208 mother board, pls check this link to download the 1.1.8 TMC2208 Ender3 pro firmware: https://www.creality3d.cn/creality3d-ender-3-pro-3d-printer-p00251p1.html "
> 
> Two different Ender 3 Pro links, https://www.creality3d.cn/creality3d-ender-3-pro-p00251p1.html or https://www.creality3d.cn/creality3d-ender-3-pro-3d-printer-p00251p1.html
> Unfortunately if you click on either link you end up with the same file: "Ender-3 Pro1.1.6.2 resume printing.hex"
> Hopefully they will get it posted soon. Meantime, if anyone has the 1.1.8 silent board for an ender 3 pro and knows how to use avrdude or whatever other means to dump a copy of the firmware, I would appreciate it. I enjoy helping folks, but I really don't want to buy a brand new board I don't need just to do that. :)

Marlin is open source.
https://github.com/MarlinFirmware/Marlin

You should definitely consider compiling it yourself. You can find the correct config file in the zip file linked on this page: https://github.com/MarlinFirmware/Marlin/tree/2.0.x/config

**Comment by cdub3d** [comment]
> Marlin is open source.
> https://github.com/MarlinFirmware/Marlin
> 
> You should definitely consider compiling it yourself. You can find the correct config file in the zip file linked on this page: https://github.com/MarlinFirmware/Marlin/tree/2.0.x/config

Oh, *I* do, every time I need it. But when a person states that they want the exact firmware they're already running, as a backup, then my compiling the firmware isn't going to help them very much. (Also, the configuration zip does not have any Ender 3 Pro config, let alone one with silent board/2208's.)
if it was one of my own, I'd be running 2.0. If I wanted a backup, I also know how to do that.

Bottom line though, is Creality should handle this. The board in question has been out for a long time. And they mention the 1.1.8 firmware for this specific board on their page, they just failed to actually provide the firmware and/or source at the link they have on the page.
